### PR TITLE
[#2769] feat(core): supports schema event for event listener

### PR DESCRIPTION
**Pull request by FANNG1**
### What changes were proposed in this pull request?
supports schema events for event listener system

### Why are the changes needed?
Fix: #2769 

### Does this PR introduce _any_ user-facing change?
no

### How was this patch tested?
existing tests

### CODE REVIEW

**Comment by FANNG1** [comment]
@jerryshao @yuqi1129 @mchades please help to review when you have time, thanks

File: core/src/main/java/com/datastrato/gravitino/listener/api/event/AlterSchemaFailureEvent.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+ * Represents an event that is triggered when an attempt to alter a schema fails due to an
+ * exception.
+ */
+@DeveloperApi
+public final class AlterSchemaFailureEvent extends SchemaFailureEvent {
+  public AlterSchemaFailureEvent(
+      String user, NameIdentifier identifier, Exception exception, SchemaChange[] schemaChanges) {
+    super(user, identifier, exception);
+  }
+}
```
Comment (jerryshao) [inline core/src/main/java/com/datastrato/gravitino/listener/api/event/AlterSchemaFailureEvent.java:22]: I think you miss the method to get `schemaChanges`.

File: core/src/main/java/com/datastrato/gravitino/listener/api/info/SchemaInfo.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+import com.datastrato.gravitino.Audit;
+import com.datastrato.gravitino.annotation.DeveloperApi;
+import com.datastrato.gravitino.rel.Schema;
+import com.google.common.collect.ImmutableMap;
+import java.util.Map;
+import javax.annotation.Nullable;
+
+@DeveloperApi
+public final class SchemaInfo {
```
Comment (jerryshao) [inline core/src/main/java/com/datastrato/gravitino/listener/api/info/SchemaInfo.java:16]: Add Javadoc to this class and method.

File: core/src/main/java/com/datastrato/gravitino/listener/api/event/AlterSchemaFailureEvent.java (hunk 1 above)
Comment (FANNG1) [reply core/src/main/java/com/datastrato/gravitino/listener/api/event/AlterSchemaFailureEvent.java:35]: done

File: core/src/main/java/com/datastrato/gravitino/listener/api/info/SchemaInfo.java (hunk 2 above)
Comment (FANNG1) [reply core/src/main/java/com/datastrato/gravitino/listener/api/info/SchemaInfo.java:17]: done

**Review by jerryshao — APPROVED** [review]

**Pull request merged**